lib/gollum/app.rb

```diff
@@ -188,7 +188,8 @@ class App < Sinatra::Base
               @etag         = page.sha
               mustache :edit
           else
-            redirect_to("/create/#{encodeURIComponent(@name)}")
+            newfile = ::File.join(@path, @name).sub(%r{^/}, '')
```

Comment:
Not sure about the `.sub` here -- what if you're trying to create a page in a subdirectoy, e.g. `Subdir/Foo`? Is the fact that the path contains slashes causing a problem?

Reply:
`Subdir/Foo` is fine because `%r{^/}` only matches a leading slash. The problem is `/Subdir/Foo`, which is redirected to `/gollum/create//Subdir/Foo`. See the next line: `to("gollum/create/#{encodeURIComponent(newfile)}")`. Removing `.sub` results in the following failure:
```
/Users/watal/fork/gollum/test/test_app.rb:308:in `block (2 levels) in <top (required)>'
<"/gollum/create/E"> expected but was
<"/gollum/create/%2FE">
```
https://github.com/gollum/gollum/blob/5a8750a9754f46d3bf76b66eb1e03051cbbb29a3/test/test_app.rb#L305-L308

Reply:
If `::File.join(@path, @name)` always starts with `/`, then the following code is much simpler:
```rb
newfile = ::File.join(@path, @name)
redirect to("gollum/create#{encodeURIComponent(newfile)}")
```

Reply:
`::File.join(@path, @name)` only starts with a `/` if `@path` starts with a `/`. If `@path` can fail to contain a `/` (or can be `nil`), maybe the following works:

```ruby
redirect to("gollum/create#{::File.join('/', @path, @name)}")
```

Or alternatively, use [this helper](https://github.com/gollum/gollum/blob/master/lib/gollum/helpers.rb#L20). That does the same as you `sub`, it's just a little easier to maintain.

Reply:
Perfect! It also indicates that replacing `%2F`with `/` has been considered safe and clean so far.